hazelcast/src/main/java/com/hazelcast/config/Config.java

```diff
@@ -207,32 +216,228 @@ public Config(String instanceName) {
+        }
+        if (path.endsWith(".yaml") || path.endsWith(".yml")) {
+            return new YamlConfigBuilder(stream).setProperties(properties).build();
+        }
+
+        throw new IllegalArgumentException("Unknown configuration file extension");
+    }
+
+    /**
+     * Creates a Config from the provided string (XML or YAML content) and uses the
+     * System.properties for variable resolution.
+     *
+     * @param source the XML or YAML content
+     * @throws IllegalArgumentException if the source is null or empty
+     * @throws com.hazelcast.core.HazelcastException if the source content is invalid
+     * @return Config created from the string
+     */
+    public static Config loadFromString(String source) {
+        return loadFromString(source, System.getProperties());
+    }
+
+    /**
+     * Creates a Config from the provided string (XML or YAML content).
+     *
+     * @param source the XML or YAML content
+     * @param properties properties to use for variable resolution
+     * @throws IllegalArgumentException if the source is null or empty
+     * @throws com.hazelcast.core.HazelcastException if the source content is invalid
+     * @return Config created from the string
+     */
+    public static Config loadFromString(String source, Properties properties) {
+        if (isNullOrEmptyAfterTrim(source)) {
+            throw new IllegalArgumentException("provided string configuration is null or empty! "
+                    + "Please use a well-structured content.");
+        }
+        Config cfg;
+        try {
+            byte[] bytes = source.getBytes();
+            InputStream stream = new ByteArrayInputStream(bytes);
+            cfg = loadFromConfigStream(new ConfigStream(stream, bytes.length), properties);
+        } catch (Exception e) {
+            throw ExceptionUtil.rethrow(e);
+        }
+        return cfg;
+    }
+
+    /**
+     * Creates a Config from the provided stream (XML or YAML content) and uses the
+     * System.properties for variable resolution.
+     *
+     * @param source the XML or YAML stream
+     * @throws com.hazelcast.core.HazelcastException if the source content is invalid
+     * @return Config created from the stream
+     */
+    public static Config loadFromStream(InputStream source) {
+        return loadFromStream(source, System.getProperties());
+    }
+
+    /**
+     * Creates a Config from the provided stream (XML or YAML content).
+     *
+     * @param source the XML or YAML stream
+     * @param properties properties to use for variable resolution
+     * @throws com.hazelcast.core.HazelcastException if the source content is invalid
+     * @return Config created from the stream
+     */
+    public static Config loadFromStream(InputStream source, Properties properties) {
+        Config cfg = null;
+        try {
+            cfg = loadFromConfigStream(new ConfigStream(source), properties);
```

Comment:
I believe we might invalidate the config here if it is more than 4K. The `ConfigStream` was designed to protect against attacks with untrusted streams, therefore we have this limitation. So I would pass the original stream down and if the `ConfigStream` wrapping it is recognized, we can build the config from the original stream, not from `ConfigStream`. Not sure if we have the same concerns with the original stream where this `Config.load()` can be used, probably not.

Alternatively, we can make the used size of the `ConfigStream` configurable via sysprops, but I think that's not really a good option.

Reply:
Thanks a lot :) It is fixed now.